### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by iluetkeb**
We should reconcile these two branches.

### CODE REVIEW

File: Installation/repos/client_minimum.repos [hunk 1]
```diff
@@ -113,6 +109,11 @@ repositories:
     version: v1.0.2
 
 # MicroROS
+
+  ros2/rclcpp:
+    type: git
+    url: https://github.com/micro-ROS/rclcpp.git
+    version: cbg-executor-0.5.1
```
Comment (iluetkeb) [inline Installation/repos/client_minimum.repos]: Why not 0.6.1?

Also, both of these are still ROS2 Crystal, I think.

Comment (BorjaOuterelo) [reply Installation/repos/client_minimum.repos]: I just used the one in master.
Shall I update it?.

Comment (BorjaOuterelo) [reply Installation/repos/client_minimum.repos]: I pushed 0.6.1 in 0a29df3.

**Comment by ralph-lange** [comment]
@BorjaOuterelo, the only package that required the callback-group-level Executor (branch cbg-executor-0.5.1 in rclcpp) was micro-ROS-demos. As the callback-group-level Executor is really experimental code only, I moved the corresponding demo to the new micro-ROS_experiments repo, which should not be included in any .repo file. Thus, if we update the .repo files in micro-ROS-doc/repos to the latest version of micro-ROS-demos, we may refer to the mainline rclcpp branch (i.e. master or tag 0.7.5) in .repo files.

**Comment by BorjaOuterelo** [comment]
@ralph-lange right, I will align our .repo with the one in https://raw.githubusercontent.com/ros2/ros2/crystal/ros2.repos and point to latest micro-ROS-demos.

**Comment by iluetkeb** [comment]
Regarding the repo file, maybe have a look at the approach I took in https://github.com/micro-ROS/micro-ros-build/tree/micro-ros-setup -- I just used the normal ros2 repo file and then removed some stuff. currently I'm doing that after checkout, because it was easier, but we could also modify the repo file on the fly.

btw, that repo is intended to simplify installation, so you could also use the script therein, instead of creating the workspace manually.

File: Installation/NuttxLinux_DevTools.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    apt install -y  remake \
+                    vim \
+                    screen \
+                    ranger \
+                    minicom
+    ```
+
+1. **Clone nuttx repo**
+    ```bash
+    git clone https://bitbucket.com/nuttx/nuttx.git ~/nuttx
```
Comment (imuguruza) [inline Installation/NuttxLinux_DevTools.md]: git clone https://github.com/micro-ROS/NuttX.git ~/nuttx

**Comment by BorjaOuterelo** [comment]
@ingo I am updating the .repos living here to use the latest crystal (found some issues compiling micro-ROS-agent currently working on them). Do you prefer me to apply these changes to the build repo?.

**Comment by iluetkeb** [comment]
> @iluetkeb I am updating the .repos living here to use the latest crystal (found some issues compiling micro-ROS-agent currently working on them). Do you prefer me to apply these changes to the build repo?.

Does the create_agent_ws.sh script work for you?

**Comment by BorjaOuterelo** [comment]
I haven't tried it yet.

**Comment by iluetkeb** [comment]
For now, I would just keep your repos file as-is, to maintain your current workflow. Once you've determined whether you can use the scripts, you can just remove them.

File: Installation/NuttxLinux_DevTools.md (hunk 2 above)
Comment (BorjaOuterelo) [inline Installation/NuttxLinux_DevTools.md]:
```suggestion
    git clone https://github.com/micro-ROS/NuttX.git ~/nuttx
```

**Pull request merged**